pycaret/time_series/forecasting/oop.py

```diff
@@ -1217,6 +1219,124 @@ def _disable_metrics(self) -> "TSForecastingExperiment":
+                "or the model is not in the allowed list of models for this setup."
+            )
+            self.logger.info(msg)
+            print(msg)
+        return engine
+
+    def set_engine(self, estimator: str, engine: str, severity="error"):
+        """Sets the engine to use for a particular model.
+
+        Parameters
+        ----------
+        estimator : str
+            Identifier for the model for which the engine should be set, e.g.
+            "auto_arima"
+        engine : str
+            Engine to set for the model. All available engines for the model
+            can be retrieved using get_allowed_engines()
+        severity : str, optional
+            How to handle incorrectly specified engines. Allowed values are "error"
+            and "warning". If set to "warning", the existing engine is left
+            unchanged if the specified engine is not correct., by default "error".
+
+        Raises
+        ------
+        ValueError
+            (1) If specified engine is not in the allowed list of engines and
+                severity is set to "error"
+            (2) If the value of "severity" is not one of the allowed values
+        """
+        allowed_engines = self.get_allowed_engines(estimator=estimator)
+        if allowed_engines is None:
+            msg = (
+                f"Either model '{estimator}' has only 1 engine and hence can not be changed, "
+                "or the model is not in the allowed list of models for this setup."
+            )
+
+            if severity == "error":
+                raise ValueError(msg)
+            elif severity == "warning":
+                self.logger.warning(msg)
+                print(msg)
+            else:
+                raise ValueError(
+                    "Error in calling set_engine, severity "
+                    f'argument must be "error" or "warning", found "{severity}".'
+                )
+        else:
+            self.model_engines[estimator] = engine
+            self.logger.info(
+                f"Engine successfully changes for model '{estimator}' to '{engine}'."
+            )
+
+        # Need to do this, else internal class variables are not reset with new engine.
+        self._set_all_models()
+
+    def _set_default_model_engines(
+        self, engines: Optional[Dict[str, str]] = None
+    ) -> "TSForecastingExperiment":
+        """Set all the default model engines
+
+        engines: Optional[Dict[str, str]] = None
+            The engine to use for the models, e.g. for auto_arima, users can
+            switch between "pmdarima" and "statsforecast" by specifying
+            engines={"auto_arima": "statsforecast"}
+
+        Returns
+        -------
+        TSForecastingExperiment
+            The experiment object to allow chaining of methods
+        """
+        # Other engines will default to sktime defaults
+        defaults = {"auto_arima": "pmdarima"}
```

Comment:
Can we make that an attribute of the container? It doesn't seem maintainable to hardcode it here